llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -824,6 +826,48 @@ void InterBlockScheduling::emitInterBlockTop(const BlockState &BS) const {
   }
 }
 
+void InterBlockScheduling::emitInterBlockTop(BlockState &BS) {
+  if (BS.Kind != BlockType::Epilogue) {
+    return;
+  }
+
+  MachineBasicBlock *BB = BS.TheBlock;
+  MachineBasicBlock *Loop = getLoopPredecessor(*BB);
+  assert(Loop);
+  const BlockState &LBS = getBlockState(Loop);
+
+  // Emit the bundles of the swp epilog in a dedicated exit.
+  // If there isn't one, spawn a new block, add a new block state and put
+  // this block to be scheduled later. Some maintenance of the original block
+  // state is also necessary.
+  if (LBS.isPipelined()) {
```

Comment:
Nit: Same, maybe make that an early exit to make it clear we only handle pipelined loops here, and non-pipelined ones are handled through a safety margin.